### PR TITLE
chore: improve single-node.sh

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://github.com/celestiaorg/celestia-app/issues/3975

Fixes a few issues I discovered with single-node.sh

1. `single-node.sh` previously created a genesis file with `appVersion: 3` so even though it attempted to upgrade from v1 -> v2, that step doesn't make any sense because the node was already on v3. This PR fixes the issue by removing the `--v2-upgrade-height` flag from the script
2. Add a few more logs
3. Add a new script: `single-node-upgrades.sh` that can be invoked to start a single node on v1 and upgrade it: v1 -> v2 -> v3. After running the command, the user will see

```
An upgrade is pending to app version 3 at height 50406.
```

## FLUPs

1. Modify single-node.sh to override the upgrade height delay and instead upgrade from v2 -> v3 instantly: https://github.com/celestiaorg/celestia-app/issues/4053

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -459,6 +459,7 @@ func (app *App) EndBlocker(ctx sdk.Context, req abci.RequestEndBlock) abci.Respo
 	res := app.manager.EndBlock(ctx, req)
 	currentVersion := app.AppVersion()
 	// For v1 only we upgrade using an agreed upon height known ahead of time
+	app.BaseApp.Logger().Debug(fmt.Sprintf("currentVersion %v", currentVersion))
```
Comment (rach-id) [inline app/app.go:462]: [nit]
why not something like:
```suggestion
	app.BaseApp.Logger().Debug("current version",  "version", currentVersion)
```

File: app/app.go [hunk 2]
```diff
@@ -476,6 +477,7 @@ func (app *App) EndBlocker(ctx sdk.Context, req abci.RequestEndBlock) abci.Respo
 	} else if shouldUpgrade, newVersion := app.SignalKeeper.ShouldUpgrade(ctx); shouldUpgrade {
 		// Version changes must be increasing. Downgrades are not permitted
 		if newVersion > currentVersion {
+			app.BaseApp.Logger().Info(fmt.Sprintf("upgrading from app version %v to %v", currentVersion, newVersion))
```
Comment (rach-id) [inline app/app.go:480]: [nit]
same
```suggestion
			app.BaseApp.Logger().Info("upgrading app version", "current version", currentVersion, "new version", newVersion))
```

File: scripts/upgrade-to-v3.sh [hunk 3]
```diff
@@ -0,0 +1,55 @@
+#!/bin/sh
+
+# Prerequisites: ensure ./single-node.sh is running in another terminal.
+# Wait until block height is 3 for the node to upgrade from v1 -> v2.
+# This script will upgrade the node from v2 -> v3.
```
Comment (rach-id) [inline scripts/upgrade-to-v3.sh:5]: [nit]

```suggestion
# This script will upgrade the node from v2 -> v3.
# Prerequisites: ensure ./single-node.sh is running in another terminal.
# Wait until block height is 3 for the node to upgrade from v1 -> v2.
```

**Review by rach-id — DISMISSED** [review]

File: app/app.go [hunk 4]
```diff
@@ -476,6 +477,7 @@
 	} else if shouldUpgrade, newVersion := app.SignalKeeper.ShouldUpgrade(ctx); shouldUpgrade {
 		// Version changes must be increasing. Downgrades are not permitted
 		if newVersion > currentVersion {
+			app.BaseApp.Logger().Info("upgrading app version", "current version", currentVersion, "new version", newVersion))
```
[the remark this answers is not in the document]
Comment (rach-id) [reply app/app.go:480]: This should be applied before merging. I made a mistake leaving an extra ) with the suggestion

**Review by evan-forbes — DISMISSED** [review]

**Review by rach-id — DISMISSED** [review]

File: app/app.go [hunk 5]
```diff
@@ -459,6 +459,7 @@ func (app *App) EndBlocker(ctx sdk.Context, req abci.RequestEndBlock) abci.Respo
 	res := app.manager.EndBlock(ctx, req)
 	currentVersion := app.AppVersion()
 	// For v1 only we upgrade using an agreed upon height known ahead of time
+	app.BaseApp.Logger().Debug("current version", "version", currentVersion)
```
Comment (cmwaters) [inline app/app.go:462]: Don't you think this is a little too spammy (I know it's debug logs) to have the version printed for each block?

**Review by cmwaters — COMMENTED** [review]
I would actually think it would be more convenient that the `single-node.sh` scripts starts from the latest version. We're no longer that concerned with v1 and soon to be v2

File: app/app.go (hunk 5 above)
Comment (rootulp) [reply app/app.go:462]: Yea fair enough, it's mostly for my debugging but prob not useful for node operators. I can remove this line and add it back when I debug version issues.

**Comment by rootulp** [comment]
> I would actually think it would be more convenient that the single-node.sh scripts starts from the latest version. We're no longer that concerned with v1 and soon to be v2

That is fine with me. I'll move the current `single-node.sh` stuff to a new script that actually starts the chain at v1, upgrades to v2, then to v3 so that we can easily test for app version edge cases.

Even though we only care about the v3 state machine for current modifications, I think it's still important to test that we can do v1 -> v2 -> v3 correctly b/c that's what mainnet nodes do.

**Review by cmwaters — APPROVED** [review]
Happy to merge this. The `single-node-upgrades.sh` works as expected. As a side note: the chain won't actually upgrade until height 50406, so this seems good just for testing that the upgrade works

**Comment by rootulp** [comment]
> Happy to merge this. The single-node-upgrades.sh works as expected. As a side note: the chain won't actually upgrade until height 50406, so this seems good just for testing that the upgrade works

https://github.com/celestiaorg/celestia-app/pull/4065 should make it upgrade from v2 -> v3 faster (like 3 blocks)

**Review by evan-forbes — APPROVED** [review]
